### PR TITLE
Cut down federated testing docs

### PR DESCRIPTION
**Pull request by spiffxp**
Much of this has become out of date and is better documented elsewhere within this repo

This also addresses #3311 by just asking for "keep it <90 days stale."  Cutting down always-failing cruft I'll tackle elsewhere, but we've got a fair number of stale results that this can help address.

### CODE REVIEW

**Comment by BenTheElder** [comment]
SGTM at a glance, might be worth mentioning one of the actual federated jobs, I also want to double check the GCS bucket requirements (cc'd relevant parties above).

File: docs/contributing-test-results.md [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+  `dashboard_tab`s)
+
+We are actively working on improving this process, which means that this
+document may not be kept exactly up-to-date. Feel free to file an issue against
+this repo when you run into problems.
+
+We prefer test results that are actively kept up to date and maintained. This is
+especially true for testgrid, where stale dashboards clutter up the UI. We may
+periodically identify GCS buckets, jobs, or testgrid dashboards that have become
+more than 90 days stale and remove them via revertable PR
```
Comment (rmmh) [inline docs/contributing-test-results.md]: "via a revertible PR."

**Comment by spiffxp** [comment]
/hold
for squash at end

File: docs/contributing-test-results.md [hunk 2]
```diff
@@ -29,4 +29,4 @@ this repo when you run into problems.
 We prefer test results that are actively kept up to date and maintained. This is
 especially true for testgrid, where stale dashboards clutter up the UI. We may
 periodically identify GCS buckets, jobs, or testgrid dashboards that have become
-more than 90 days stale and remove them via revertable PR
+more than 90 days stale and remove them via revertible PR
```
Comment (rmmh) [inline docs/contributing-test-results.md]: period at end

**Comment by BenTheElder** [comment]
README.md still links to the old docs.

**Pull request merged**